### PR TITLE
Update ib_broker.py

### PR DESCRIPTION
**Pull request by peterfabakker**
Warn for not able to subscribe to asset

### CODE REVIEW

**Comment by tibkiss** [comment]
Please change the commit message. Instead of showing what has changed please detail why the change was created. I'd prefer to stick to scipy's conventions:
https://docs.scipy.org/doc/numpy/dev/gitwash/development_workflow.html

Thank you.

File: zipline/gens/brokers/ib_broker.py [hunk 1]
```diff
@@ -500,12 +510,21 @@ def subscribed_assets(self):
...
+            log.debug("Subscribing to market data for {}".format(
+                asset))
+
             # remove str() cast to have a fun debugging journey
             self._tws.subscribe_to_market_data(str(asset.symbol))
             self._subscribed_assets.append(asset)
-
-            while asset.symbol not in self._tws.bars:
+	    counter = 0
+            while asset.symbol not in self._tws.bars and counter < _max_wait_subscribe: # waiting X secs
```
Comment (tibkiss) [inline zipline/gens/brokers/ib_broker.py]: Please use the 'polling' library for this purpose. It would make the clean much cleaner:
https://pypi.python.org/pypi/polling/0.3.0

File: zipline/gens/brokers/ib_broker.py [hunk 2]
```diff
@@ -673,11 +694,16 @@ def order(self, asset, amount, style):
             order.m_orderType = "MKT"
         elif isinstance(style, LimitOrder):
             order.m_orderType = "LMT"
+            order.m_lmtPrice = limit_price
```
Comment (tibkiss) [inline zipline/gens/brokers/ib_broker.py]: I'm not sure if it is needed.  Please don't mix two different things in one PR.

File: zipline/gens/brokers/ib_broker.py [hunk 3]
```diff
@@ -807,14 +833,14 @@ def _update_from_order_status(zp_order, ib_order_id):
                 open_order_state = self._tws.open_orders[ib_order_id]['state']
 
                 zp_status = self._ib_to_zp_status(open_order_state.m_status)
-                if zp_status:
-                    zp_order.status = zp_status
-                else:
+                if zp_status is None:
```
Comment (tibkiss) [inline zipline/gens/brokers/ib_broker.py]: Not sure why is this changed

File: zipline/gens/brokers/ib_broker.py (hunk 2 above)
Comment (peterfabakker) [reply zipline/gens/brokers/ib_broker.py]: I don't know how to split a request. This change is needed as now you cannot post limit orders, it will fail on the absence of order.lmtPrice, is a bug

File: zipline/gens/brokers/ib_broker.py (hunk 1 above)
Comment (peterfabakker) [reply zipline/gens/brokers/ib_broker.py]: I'll look at it. could be cleaner indeed

File: zipline/gens/brokers/ib_broker.py (hunk 3 above)
Comment (peterfabakker) [reply zipline/gens/brokers/ib_broker.py]: from your TAO branch, just took it over

File: zipline/gens/brokers/ib_broker.py [hunk 4]
```diff
@@ -16,7 +16,7 @@
 from time import sleep
 from math import fabs
 
-from six import iteritems
+from six import iteritems, itervalues
```
Comment (peterfabakker) [inline zipline/gens/brokers/ib_broker.py]: from your TAIO branch

File: zipline/gens/brokers/ib_broker.py (hunk 3 above)
Comment (tibkiss) [reply zipline/gens/brokers/ib_broker.py]: Please cherry-pick instead of copy paste. thanks.

File: zipline/gens/brokers/ib_broker.py (hunk 2 above)
Comment (tibkiss) [reply zipline/gens/brokers/ib_broker.py]: I presume you mean 'PR' by request. 
You need to roll back this change, create a new branch, commit with a reasonable message and post a new pr.

**Comment by tibkiss** [comment]
Taken a look on CI: five testcases are failing, plus flake8 problems are present.
Please address those as well.

My recommendation would be that we split this up into smaller chunks, where one PR fixes one problem.

File: zipline/gens/brokers/ib_broker.py (hunk 3 above)
Comment (namato) [reply zipline/gens/brokers/ib_broker.py]: hi guys, there's an additional check here a few lines below line 836 that also needs the same fix (because of the use of custom enum()s they start at 0 and thus evaluate to 0)

**Comment by tibkiss** [comment]
This change became outdated. Closing.
